### DISCUSSION TITLE
add-on library for more of libc

### DISCUSSION THREAD

**Discussion by gneverov** [picolibc/picolibc#884] [Show and tell]
I just wanted to share a project I've been working on: Morelibc. It is an add-on to Picolibc that provides more of the standard C/POSIX programming environment, particularly filesystems, threading, and dynamic linking. I am hoping others might find it a useful building block for more complex projects. It is designed to be cross-platform, but only RP Pico is currently supported. Your feedback is very much welcomed. Thanks. Link below.

https://github.com/gneverov/morelibc

**Comment by keith-packard** [comment]
Very cool!

**Comment by quic-seaswara** [comment]
Is there a plan to add support for dynamic linking in picolibc ?

**Reply by keith-packard** [reply]
I don't know of anyone with such plans

**Reply by quic-k** [reply]
@keith-packard 
are you open to dynamic linking patches in Picolibc?

**Reply by keith-packard** [reply]
Happy to merge what people want as long as it doesn't cause problems for the usual embedded targets. Note that picolibc can now act as a Linux C library, but is limited to static linking...
